### PR TITLE
Adding torchaudio recipe

### PR DESCRIPTION
**Pull request by tyrius02**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tyrius02** [comment]
@conda-forge/help-python @conda-forge/help-python-c @hmaarrfk I think this is ready to go!

Yes, I'm casting a little bit of a wide net. I've contributed other recipes in the past but the PyTorch stuff is unlike anyything else I've done.

This initial recipe targets PyTorch 1.8.0 because I, at some point, want to align w/ PyTorch 1.8.2 (the last LTS release).

When I get to torchaudio 0.10.0 and PyTorch 1.10.0 I have to split into two packages: torchaudio-gpu and torchaudio-cpu. I _think_ I know how to do this but first, I just want to get this recipe in.

File: recipes/torchaudio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{% set version = "0.8.0" %}
+{% set pytorch_version = "1.8.0" %}
```
Comment (hmaarrfk) [inline recipes/torchaudio/meta.yaml:2]: pytorch 1.8 is very likely broken at conda-forge....

It isn't that I don't want to get your recipe in. but I think that it should target the latest pinning of pytorch, 1.12 and very likely 1.13 (soon).

if this is for you trying to get your work done, I would really suggest just uploading to your own channel. 


Once the recipe is ready for wider distribution, then you can revive it here and others can benefit, but at this stage, we would be releasing software on day 1, with no real path to installation.

**Review by hmaarrfk — CHANGES_REQUESTED** [review]

**Comment by hmaarrfk** [comment]
If you want to revive an LTS branch of pytorch, we should do that in the feedstock first.

https://github.com/conda-forge/pytorch-cpu-feedstock/

**Comment by hmaarrfk** [comment]
I have to say, pytorch is a giant pain to build because a member of conda-forge/core must upload it to anaconda manually. As such, it won't be easy to support multiple branches.

I personally think that I would would not have time to approve PRs to a 1.8 branch on the grounds that I don't have time to upload them. An other member of core might upload them, but in reality, they are also pretty busy. 

There was some discussion on splitting the package, but I got through the CPU portions, and then got tired. https://github.com/conda-forge/pytorch-cpu-feedstock/issues/108

without interest from upstream cleanly splitting their project up, this is somewhat dead in the water.

**Pull request closed without merging**